plugins/Kaleidoscope-DynamicMacros/src/kaleidoscope/plugin/DynamicMacros.cpp

```diff
@@ -272,14 +275,14 @@ EventHandlerResult DynamicMacros::onFocusEvent(const char *command) {
     } else {
       uint16_t pos = 0;
 
-      while (!::Focus.isEOL()) {
+      while (!::Focus.isEOL() && pos < storage_size_) {
```

Comment:
If we're counting on Chrysalis/whatever to be well-behaved, we don't need this check.  If not, I think it would be better to not do the commit (I assume that's where the actual write to EEPROM takes place) if the writer tried to overflow the storage slice.

Reply:
On ARM, yes, commit is where the actual write takes place. On AVR, we write immediately, and commit is a no-op. So truncating is consistent with AVR, and also with other similar plugins (colormap, eeprom-keymap).

Reply:
So, the guard here stops a buggy version of Chrysalis from messing up parts of the storage outside the slice, so I want to retract my earlier statement about not needing the check; it's still useful even if we aren't doing other things to stop it from causing our macros to read out of bounds.